src/Analyzer/Passes/CrossToInnerJoinPass.cpp

```diff
@@ -0,0 +1,253 @@
+#include <Analyzer/Passes/CrossToInnerJoinPass.h>
+
+#include <Analyzer/InDepthQueryTreeVisitor.h>
+
+#include <Analyzer/JoinNode.h>
+#include <Analyzer/QueryNode.h>
+#include <Analyzer/FunctionNode.h>
+#include <Analyzer/ConstantNode.h>
+#include <Analyzer/ColumnNode.h>
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+
+#include <Common/logger_useful.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int INCORRECT_QUERY;
+}
+
+namespace
+{
+
+using EquiCondition = std::tuple<QueryTreeNodePtr, QueryTreeNodePtr>;
+
+void exctractJoinConditions(const QueryTreeNodePtr & node, QueryTreeNodes & equi_conditions, QueryTreeNodes & other)
+{
+    if (auto * func = node->as<FunctionNode>())
```

Comment:
Consider to rewrite this to improve readability.
If node is not function node we can just add node to other and return at the top of the function:
```
other.push_back(node);
return;
```